### PR TITLE
Upload file command with copy/paste logic

### PR DESCRIPTION
**Pull request by prcrst**
It works, I'd like to get rid of some of the copy/pasted code

### CODE REVIEW

File: plugs/editor/upload.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    if (file.content.length > maximumAttachmentSize) {
+        editor.flashNotification(
+            `Attachment is too large, maximum is ${maximumAttachmentSize / 1024 / 1024
+            }MB`,
+            "error",
+        );
+        return;
+    }
+
+    const suggestedName = resolve(folderName(await editor.getCurrentPage()), file.name);
```
Comment (zefhemel) [inline plugs/editor/upload.ts:40]: Honestly I think you can replace the `resolve` call with a simple `folderName(...) + "/" + file.name` here and get rid of the fancy implementation of that.

Comment (prcrst) [reply plugs/editor/upload.ts:40]: ok

File: plug-api/types.ts [hunk 2]
```diff
@@ -143,5 +143,6 @@ export type LintDiagnostic = {
 
 export type UploadFile = {
   name: string;
+  type: string;
```
Comment (zefhemel) [inline plug-api/types.ts:146]: and maybe rename this to `contentType` for consistency with other parts of the code

Comment (prcrst) [reply plug-api/types.ts:146]: ok, I kept it consistent with `File`... I'll change it

**Comment by zefhemel** [comment]
Pushed one minor tweak (for when uploading to the root folder). Merging!

**Pull request merged**